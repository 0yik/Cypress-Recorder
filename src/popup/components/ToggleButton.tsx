import * as React from 'react';
import { ControlAction } from '../../constants';

export interface ToggleButtonProps {
  recStatus: String,
  handleToggle: (action: ControlAction) => void,
}

export default ({ recStatus, handleToggle }: ToggleButtonProps) => {
  const handleClick = (): void => {
    let action: ControlAction | null = null;
    if (recStatus === 'off') action = ControlAction.START;
    else if (recStatus === 'on') action = ControlAction.STOP;
    else if (recStatus === 'done') action = ControlAction.RESET;
    if (action) {
      handleToggle(action);
    }
  };

  return (
    <>
      <button className="button" onClick={handleClick}>
        {recStatus === 'off'
          ? 'Start Recording'
          : recStatus === 'on'
            ? 'Stop Recording'
            : 'Reset Recording'}
      </button>
    </>
  );
}
