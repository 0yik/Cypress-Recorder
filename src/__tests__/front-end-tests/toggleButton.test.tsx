import * as React from 'react';
import ToggleButton from '../../popup/components/ToggleButton';
import { shallow } from "enzyme";
import "../../setupTests"

describe('toggle button', () => {
  let wrapper;
  const handleToggleMock = jest.fn()
  it('should call handletoggle with argument "startRec" when clicked and recStatus is "off"', () => {
    wrapper = shallow(<ToggleButton recStatus='off' handleToggle={handleToggleMock}/>)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith('startRec')
  })
  it('should call handletoggle with argument "stopRec" when clicked and recStatus is "on"', () => {
    wrapper = shallow(<ToggleButton recStatus='on' handleToggle={handleToggleMock}/>)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith('stopRec');
  })
  it('should call handletoggle with argument "resetRec" when clicked and recStatus is "done"', () => {
    wrapper = shallow(<ToggleButton recStatus='done' handleToggle={handleToggleMock}/>)
    expect(wrapper).toMatchSnapshot();
    const button = wrapper.find('button');
    button.simulate('click');
    expect(handleToggleMock).toHaveBeenCalledWith('resetRec')
  })
})
