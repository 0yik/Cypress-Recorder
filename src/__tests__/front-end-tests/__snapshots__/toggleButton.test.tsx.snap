// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`toggle button should call handletoggle with argument "startRec" when clicked and recStatus is "off" 1`] = `
<Fragment>
  <button
    className="button"
    onClick={[Function]}
  >
    Start Recording
  </button>
</Fragment>
`;

exports[`toggle button should render correct text when rec status is done 1`] = `
<Fragment>
  <button
    className="button"
    onClick={[Function]}
  >
    Reset Recording
  </button>
</Fragment>
`;

exports[`toggle button should render correct text when rec status is on 1`] = `
<Fragment>
  <button
    className="button"
    onClick={[Function]}
  >
    Stop Recording
  </button>
</Fragment>
`;
